src/pkg/artifact/model/model.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"time"
+
+	"github.com/goharbor/harbor/src/common/models"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// Artifact is the abstract object managed by Harbor. It hides the
+// underlying concrete detail and provides an unified artifact view
+// for all users.
+type Artifact struct {
+	ID                    int64                      `json:"id"`
+	Type                  string                     `json:"type"` // image, chart, etc
+	Repository            *models.RepoRecord         `json:"repository"`
+	Tags                  []*Tag                     `json:"tags"`       // the list of tags that attached to the artifact
+	MediaType             string                     `json:"media_type"` // the specific media type
+	Digest                string                     `json:"digest"`
+	Size                  int64                      `json:"size"`
+	UploadTime            time.Time                  `json:"upload_time"`
+	Labels                []*models.Label            `json:"labels"`
+	Signature             *Signature                 `json:"signature"`               // add the signature in the artifact level rather than tag level as we cannot make sure the signature always apply to tag
+	ExtraAttrs            map[string]interface{}     `json:"extra_attrs"`             // only contains the simple attributes specific for the different artifact type, most of them should come from the config layer
+	AdditionResourceLinks map[string][]*ResourceLink `json:"addition_resource_links"` // the resource link for build history(image), scan result(image), values.yaml(chart), dependency(chart), etc
+	Annotations           map[string]string          `json:"annotations"`
+	References            []*Reference               `json:"references"` // child artifacts referenced by the parent artifact if the artifact is an index
+}
+
+// ResourceLink is a link via that a resource can be fetched
+type ResourceLink struct {
+	HREF string `json:"href"`
```

Comment:
Will we potentially support relative and absolute URIs?
so probably add a flag or an attribute called `type`?

Reply:
Updated.